core/controllers/base_test.py

```diff
@@ -95,16 +114,72 @@ def test_requests_for_invalid_paths(self):
         response = self.testapp.put('/library/extra', {}, expect_errors=True)
         self.assertEqual(response.status_int, 404)
 
-    def test_redirect_in_both_logged_in_and_logged_out_states(self):
-        "Test for a redirect in both logged in and logged out states on '/'."
+    def test_redirect_in_logged_out_states(self):
+        """Test for a redirect in logged out state on '/'."""
 
-        # Logged out state
         response = self.testapp.get('/')
         self.assertEqual(response.status_int, 302)
         self.assertIn('splash', response.headers['location'])
 
-        # Login and assert that there is a redirect
-        self.login('user@example.com')
+    def test_root_redirect_rules_for_logged_in_learners(self):
+        self.login(self.TEST_LEARNER_EMAIL)
+
+        # Since no explorations have been created, going to '/' should redirect
+        # to the library page.
+        response = self.testapp.get('/')
+        self.assertEqual(response.status_int, 302)
+        self.assertIn('library', response.headers['location'])
+        self.logout()
+
+    def test_root_redirect_rules_for_logged_in_creators(self):
+        self.login(self.TEST_CREATOR_EMAIL)
+        creator_user_id = self.get_user_id_from_email(self.TEST_CREATOR_EMAIL)
+        exploration_id = '0_en_test_exploration'
+        self.save_new_valid_exploration(
+            exploration_id, creator_user_id, title='Test',
+            category='Test', language_code='en')
+
+        # Since atleast one explorations have been created, going to '/' should
+        # redirect to the dashboard page.
+        response = self.testapp.get('/')
+        self.assertIn('dashboard', response.headers['location'])
+        exp_services.delete_exploration(creator_user_id, exploration_id)
+        self.logout()
+        self.login(self.TEST_CREATOR_EMAIL)
+        response = self.testapp.get('/')
+
+        # Must be on dashboard even though exploration is deleted because even
```

Comment:
Even though the exploration is deleted, the user is still redirected to the dashboard. This is because deleted explorations are still associated with their creators.